homeassistant/components/voip/voip.py

```diff
@@ -0,0 +1,179 @@
+"""Voice over IP (VoIP) implementation."""
+import asyncio
+import logging
+import time
+
+import async_timeout
+from voip_utils import CallInfo, RtpDatagramProtocol, SdpInfo, VoipDatagramProtocol
+
+from homeassistant.components import stt, tts
+from homeassistant.components.voice_assistant import (
+    Pipeline,
+    PipelineEvent,
+    PipelineEventType,
+    async_pipeline_from_audio_stream,
+)
+from homeassistant.components.voice_assistant.vad import VoiceCommandSegmenter
+from homeassistant.const import __version__
+from homeassistant.core import HomeAssistant
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HassVoipDatagramProtocol(VoipDatagramProtocol):
+    """HA UDP server for Voice over IP (VoIP)."""
+
+    def __init__(self, hass: HomeAssistant, allow_ips: set[str]) -> None:
+        """Set up VoIP call handler."""
+        super().__init__(
+            sdp_info=SdpInfo(
+                username="homeassistant",
+                id=time.monotonic_ns(),
```

Comment:
Better use a ulid from until/ulid.py

Reply:
Do you think a random integer would be a better options?

Reply:
this is fine then.